diff -r -u ../../../agon-vdp/video/envelopes/frequency.h vdp-quark104/envelopes/frequency.h
--- ../../../agon-vdp/video/envelopes/frequency.h	2023-09-25 14:28:58.718795322 +0100
+++ vdp-quark104/envelopes/frequency.h	2023-09-28 22:21:14.217364790 +0100
@@ -12,8 +12,8 @@
 
 class FrequencyEnvelope {
 	public:
-		virtual uint16_t getFrequency(uint16_t baseFrequency, uint32_t elapsed, int32_t duration);
-		virtual bool isFinished(uint32_t elapsed, int32_t duration);
+		virtual uint16_t getFrequency(uint16_t baseFrequency, uint32_t elapsed, int32_t duration) = 0;
+		virtual bool isFinished(uint32_t elapsed, int32_t duration) = 0;
 };
 
 struct FrequencyStepPhase {
diff -r -u ../../../agon-vdp/video/envelopes/volume.h vdp-quark104/envelopes/volume.h
--- ../../../agon-vdp/video/envelopes/volume.h	2023-09-25 14:28:58.718795322 +0100
+++ vdp-quark104/envelopes/volume.h	2023-09-28 22:18:40.983884819 +0100
@@ -9,9 +9,9 @@
 
 class VolumeEnvelope {
 	public:
-		virtual uint8_t getVolume(uint8_t baseVolume, uint32_t elapsed, int32_t duration);
-		virtual bool isReleasing(uint32_t elapsed, int32_t duration);
-		virtual bool isFinished(uint32_t elapsed, int32_t duration);
+		virtual uint8_t getVolume(uint8_t baseVolume, uint32_t elapsed, int32_t duration) = 0;
+		virtual bool isReleasing(uint32_t elapsed, int32_t duration) = 0;
+		virtual bool isFinished(uint32_t elapsed, int32_t duration) = 0;
 };
 
 class ADSRVolumeEnvelope : public VolumeEnvelope {
diff -r -u ../../../agon-vdp/video/video.ino vdp-quark104/video.ino
--- ../../../agon-vdp/video/video.ino	2023-09-28 21:13:54.781154250 +0100
+++ vdp-quark104/video.ino	2023-09-28 22:24:17.740720977 +0100
@@ -93,6 +93,9 @@
 	bool cursorState = false;
 
 	while (true) {
+ 		if ((count & 0x7f) == 0) delay(1 /* -TM- ms */);
+ 		count++;
+
 		if (terminalMode) {
 			do_keyboard_terminal();
 			continue;
@@ -112,7 +115,6 @@
 			count = -1;
 			processor->processNext();
 		}
-		count++;
 	}
 }
 
