diff -u ../../video/video.ino vdp-quark103/video.ino
--- ../../video/video.ino	2023-07-18 21:57:30.016111556 +0100
+++ vdp-1.03/video.ino	2023-09-13 14:12:15.805128203 +0100
@@ -135,6 +135,9 @@
 	bool cursorState = false;
 
 	while(true) {
+		count++;
+		if ((count & 0x7f) == 0) delay(1 /* -TM- ms */);
+
 		if(terminalMode) {
 			do_keyboard_terminal();
 			continue;
@@ -163,7 +166,6 @@
 			setRTSStatus(true);
 		}
 		#endif
-    	count++;
   	}
 }
 
@@ -279,7 +281,7 @@
 	int channel = *(int *)parameters;
 
 	audio_channels[channel] = new audio_channel(channel);
-	while(true) {
+	task_loop {
 		audio_channels[channel]->loop();
 		vTaskDelay(1);
 	}
